edx_notifications/stores/sql/store_provider.py

```diff
@@ -554,3 +557,20 @@ def get_all_namespaces(self, start_datetime=None, end_datetime=None):
         result_set = result_set.values_list('namespace', flat=True).order_by('namespace').distinct()
 
         return result_set
+
+    def purge_notifications_for_users(self, user_ids):
+        """
+        This will remove all notifications and preferences for given user IDs
+        """
+        user_notifications = SQLUserNotification.objects.filter(user_id__in=user_ids)
+        archived_notifications = SQLUserNotificationArchive.objects.filter(user_id__in=user_ids)
+        SQLNotificationMessage.objects.filter(id__in=user_notifications.values_list('msg_id', flat=True)).delete()
+        SQLNotificationMessage.objects.filter(id__in=archived_notifications.values_list('msg_id', flat=True)).delete()
```

Comment:
How large can these deletes grow to? Also, is this doing a subquery, or is it piping the whole query through the Python layer?

Reply:
@viadanna @xitij2000 can you please address the above queries?

Reply:
These queries can get quite large, depending on the number of users being deleted, e.g. removing 10.000 users from a large instance.

Also, there's no subquery for passing the message ids.

Reply:
@viadanna: Can you please try modifying it to use `values` instead of `values_list`? I think that should cause it to do the subquery in the database instead of passing back and forth through the Python layer:

```python
        SQLNotificationMessage.objects.filter(id__in=user_notifications.values('msg_id')).delete()
        SQLNotificationMessage.objects.filter(id__in=archived_notifications.values('msg_id')).delete()
```

We've had issues in the past with serializing large sets of these back and forth through the webapp and doing all the work in the database helped with those issues in the past. If this doesn't work for some reason, I'm okay with merging this as is. But please try it.

Thank you.

Reply:
@viadanna Can you please look into the above suggestion?

Reply:
@ormsbee @ihtram Alright, moved to using `values` instead of `values_list`